packages/atomic/src/components/commerce/atomic-commerce-product-list/atomic-commerce-product-list.tsx

```diff
@@ -218,31 +255,40 @@ export class AtomicCommerceProductList
 
   private renderAsGrid() {
     return this.productState.products.map((product, i) => {
-      const propsForAtomicProduct = this.getPropsForAtomicProduct(product);
+      const {interactiveProduct, ...restOfPropsForAtomicProduct} =
+        this.getPropsForAtomicProduct(product, i);
+      const analyticsWarning = interactiveProduct.warning;
       return (
         <DisplayGrid
           item={{
             ...product,
             clickUri: product.clickUri,
             title: product.ec_name ?? 'temp',
           }}
-          // TODO KIT-3149: add back once the interactive result is implemented
-          //{...propsForAtomicProduct.interactiveResult}
-          // TODO KIT-3149: Remove these back once the interactive result is implemented
+          {...interactiveProduct.controller}
           setRef={(element) =>
             element && this.productListCommon.setNewResultRef(element, i)
           }
-          select={function (): void {
-            throw new Error('Function not implemented. TODO KIT-3149');
+          select={() => {
+            analyticsWarning
+              ? this.logWarning(analyticsWarning)
+              : interactiveProduct.controller?.select();
           }}
-          beginDelayedSelect={function (): void {
-            throw new Error('Function not implemented. TODO KIT-3149');
+          beginDelayedSelect={() => {
+            analyticsWarning
+              ? this.logWarning(analyticsWarning)
+              : interactiveProduct.controller?.beginDelayedSelect();
           }}
-          cancelPendingSelect={function (): void {
-            throw new Error('Function not implemented. TODO KIT-3149');
+          cancelPendingSelect={() => {
+            analyticsWarning
+              ? this.logWarning(analyticsWarning)
+              : interactiveProduct.controller?.cancelPendingSelect();
           }}
         >
-          <atomic-product {...this} {...propsForAtomicProduct}></atomic-product>
```

Comment:
@fpbrault was there any legitimate reason to pass {...this} to the atomic-product?

Reply:
I took this from atomic-result/atomic-result-list, but I'm looking at it and can't find anything that wouldn't be in propsForAtomicProduct or already in AtomicProduct..!